receiver/k8sclusterreceiver/watcher.go

```diff
@@ -331,7 +344,10 @@ func (rw *resourceWatcher) onAdd(obj any) {
 }
 
 func (rw *resourceWatcher) hasDestination() bool {
-	return len(rw.metadataConsumers) != 0 || rw.entityLogConsumer != nil
+	rw.mu.RLock()
+	has := len(rw.metadataConsumers) != 0 || rw.entityLogConsumer != nil
+	rw.mu.RUnlock()
```

Comment:
These changes are extra concurrency guardrails unrelated to the PR. Should be done separately.